update/Pos/Update/DB/Misc.hs

```diff
@@ -22,3 +25,12 @@ affirmUpdateInstalled h = miscPutBi (updateTrackKey h) ()
 
 updateTrackKey :: Hash Raw -> ByteString
 updateTrackKey h = "updinst/" <> encodeUtf8 (sformat hashHexF h)
+
+lastInstallerKey :: ByteString
+lastInstallerKey = "updlast/lastDownloaded" 
+
+getLastInstallerHash :: MonadDBRead m => m (Hash Raw)
+getLastInstallerHash = fromJust <$> (miscGetBi lastInstallerKey)
```

Comment:
I think `fromJust` here is a bad idea, I would return `Maybe` instead, you can't know for sure that there is a value for this key. And in `launcher` if `getLastInstallerHash` returns `Nothing` you can assume that your installer is definitely invalid (maybe someone copied some file there).